src/test/scala/strawman/collection/test/BuildFromTest.scala

```diff
@@ -0,0 +1,116 @@
+package strawman
+package collection.test
+
+import org.junit.Test
+import strawman.collection.Iterable
+import strawman.collection.mutable.Builder
+import strawman.collection._
+
+import scala.{Any, Boolean, Either, Int, Left, None, Option, PartialFunction, Right, Some, Unit}
+import scala.Predef.{ArrowAssoc, implicitly}
+import scala.math.Ordering
+import java.lang.String
+
+class BuildFromTest {
```

Comment:
Note that this is not required, these examples would work as well as follows:

~~~ scala
// def optionSequence[A](xs: Iterable[Option[A]])(implicit bf: BuildFrom[xs.type, A]): Option[bf.To]
def optionSequence[A, CC[X] <: Iterable[X]](xs: CC[Option[A]])(implicit bf: BuildFrom[CC[Option[A]], A]): Option[bf.To]
~~~

Also, users can choose to use a path dependent type (`bf.To`) or to take an additional `To` type parameter (by using `BuildFrom.Aux` -- which I could rename to something else).